### PR TITLE
[lua] Remove hardcoded ids from Bibiki Bay

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

After a recent shift, most hardcoded mob (and npc) ids in Bibiki bay would have been made useless. This moves them all over to a pattern where we ask the zone for the entities by name and their relative index in the zone compared to other entities of the same name in that zone. These lookups protect these PH's, qm's, etc. from breaking during future shifts.

## Steps to test these changes

See comments

### CODE REVIEW

**Comment by zach2good** [comment]
lol, update hasn't even landed yet...

File: scripts/zones/Bibiki_Bay/mobs/Jagil.lua [hunk 1]
```diff
@@ -12,7 +11,14 @@ entity.onMobDeath = function(mob, player, optParams)
 end
 
 entity.onMobDespawn = function(mob)
-    xi.mob.phOnDespawn(mob, ID.mob.SERRA_PH, 10, 3600) -- 1 hour
+    -- PH is the 5th Jagil entry in the zone
+    local jagilId  = mob:getZone():queryEntitiesByName('Jagil')[5]:getID()
+    local shenId   = mob:getZone():queryEntitiesByName('Shen')[1]:getID()
```
Comment (TeoTwawki) [inline scripts/zones/Bibiki_Bay/mobs/Jagil.lua]: Shen is not related to this PH or its NM

Comment (zach2good) [reply scripts/zones/Bibiki_Bay/mobs/Jagil.lua]: Good catch 👍

**Comment by kaincenteno** [comment]
nice so now we are quering by name instead of placing the npc ID inside the zone IDs?

**Comment by TeoTwawki** [comment]
> Remove hardcoded ids from Bibiki Bay 

That is what the man said, Kain. 

Lookup by name was added awhile back and is gradually getting used more to avoid shifting IDs.

**Comment by zach2good** [comment]
Tested by:
- Killing Shen and his filtrates: no errors
- Kill each of the PH mobs: no errors

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**